dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/directory/XdsDirectory.java

```diff
@@ -0,0 +1,165 @@
+package org.apache.dubbo.rpc.cluster.directory;
+
+import org.apache.dubbo.common.URL;
+import org.apache.dubbo.common.utils.CollectionUtils;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.Protocol;
+import org.apache.dubbo.rpc.cluster.Directory;
+import org.apache.dubbo.rpc.cluster.RouterChain;
+import org.apache.dubbo.rpc.cluster.SingleRouterChain;
+import org.apache.dubbo.rpc.cluster.router.state.BitList;
+import org.apache.dubbo.rpc.cluster.xds.PilotExchanger;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsCluster;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsClusterWeight;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsEndpoint;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsRoute;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsRouteAction;
+import org.apache.dubbo.rpc.cluster.xds.resource.XdsVirtualHost;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class XdsDirectory<T>  extends AbstractDirectory<T>{
+
+    private final URL url;
+
+    private final Class<T> serviceType;
+
+    private final String applicationName;
+
+    private final String protocolName;
+
+    PilotExchanger pilotExchanger = PilotExchanger.getInstance();
+
+    private Set<String> subscribeApplications;
+
+    protected RouterChain<T> routerChain;
+
+    protected List<Invoker<T>> invokers;
+    private Protocol protocol;
+
+    private final Map<String, XdsVirtualHost> xdsVirtualHostMap = new ConcurrentHashMap<>();
+
+    private final Map<String, XdsCluster<T>> xdsClusterMap = new ConcurrentHashMap<>();
```

Comment:
What's this field designed and used for?

Reply:
Retain the resource copy in the XdsRouter for the time being to check whether it improves performance.